### ISSUE TITLE
centos 6 - glibc requirements not met error

### ISSUE THREAD

**Issue by 15tolife** [GameServerManagers/LinuxGSM#2761]
**Info:**
Hy ,  please have patience , I dont know if I posted where I should , I did not sleep for 3 days trying to solve this issue....I tryed everithing i could find on google , i dont know what am I missing.
-Specs :VPS   single core  , 3gb ram , 75gb ssd , Centos6.10-x86_64 
-Reason for centos6 : my forum was made in 2011 and works only with php 5.6 , upgrading to centos 7 out of question.

**Problem:**
-CS 1.6 and CSGO both installed through LGSM both working but...every time I stop start restart i get this errors  and it makes me think that some plugins aren't working because of that , like for example something simple like dproto is just failing to load.
as you can see python 3 is installed and running and glibc 2.17 available but i dont know how to show the program to use it....
I want to mention that I have no knowledge of linux since a week ago when i bought this vps and got everithing installed and running through trial and error with help of allot of tutorials.
Basically I have no more energy to do this and as a last resort im asking for help.
Thanks in advance!
errors----> 
![cs restart](https://user-images.githubusercontent.com/62301706/76881888-8125d980-6882-11ea-89bf-48273d90ed4c.jpg)
![python and glibc](https://user-images.githubusercontent.com/62301706/76881894-82570680-6882-11ea-9716-a378405f878d.jpg)

**Comment by h3o66** [comment]
> -Reason for centos6 : my forum was made in 2011 and works only with php 5.6 , upgrading to centos 7 out of question.

Well not a reason to me :thinking: 
CentOS 7 has php5.4 as default in the repos (and remi php56 is also there for centos 7).
Only if you use other repos, that upgarde php, are the problem.

I guess you only dont want to upgrade your server.


> as you can see python 3 is installed and running and glibc 2.17 available but i dont know how to show the program to use it....

wel for a cs 1.6 server you need to upgrade the i686 version of glibc then and not the x64_86 version of the lib.

**Comment by 15tolife** [comment]
Thank you for your answer , I had centos 7 on my vps and the forum just gave 500 errors , that fixed with the downgrade to centos 6.
Can you please tell me how to upgrade to i686 ?
thank you

**Comment by h3o66** [comment]
> Can you please tell me how to upgrade to i686 ? and what about python ?
The x32 bit glibc is there only with version 2.12 installed, so you need to upgarde it.
However you did this with the 64 bit version.

**Comment by 15tolife** [comment]
a link or a list of commands of how to do that would be helpfull

**Comment by h3o66** [comment]
this could help (Not Tested):
https://gist.github.com/harv/f86690fcad94f655906ee9e37c85b174
That references this repo:
https://copr.fedorainfracloud.org/coprs/mosquito/myrepo-el6/
if you only add the repo and updte it could work.

Fyi: CentOS 6 will be only support till November this year so upgrading might be still something to be considerd

**Comment by dgibbs64** [comment]
Upgrading glibc can be risky. I wouldnt do it on a production server

**Comment by dgibbs64** [comment]
If you can't upgrade your server I would get a vps specifics for your game server

**Comment by h3o66** [comment]
> Upgrading glibc can be risky. I wouldnt do it on a production server

yeah - but if he wants to runs this game server he needs to do it ...
Or to Upgrade the system, but i mentioned that now 3 times :D

**Comment by 15tolife** [comment]
this is what i get from installing  glibc-2.17-55.el6.src.rpm
warning: user mockbuild does not exist - using root
warning: group mockbuild does not exist - using root
warning: user mockbuild does not exist - using root
warning: group mockbuild does not exist - using root
warning: user mockbuild does not exist - using root
warning: group mockbuild does not exist - using root

**Comment by 15tolife** [comment]
Thanks so much it worked no more glibc error , but I still get this  Missing dependencies: util-linux python3 nmap-ncat

**Comment by h3o66** [comment]
well, you need to install the "normal" build rpm and not the source rpm.
with the "src.rpm" you would need to build it yourself.

The easiest would be to add the repo i mentioned and just update the components via yum

**Comment by 15tolife** [comment]
can you tell me how to solve this ? missing dependencies util-linux python3 nmap-ncat

**Comment by h3o66** [comment]
> can you tell me how to solve this ? missing dependencies util-linux python3 nmap-ncat

sry - no i do not have an idea to to solve the other things.
The server could start now, but be aware that lgsm might not support centos 6 to long.

**Comment by 15tolife** [comment]
So for anyone having this problem this is what you need to do : 
-first verify what glibc version you have : 
**ldd --version**
-if you have .i686  then go here  https://download.copr.fedorainfracloud.org/results/mosquito/myrepo-el6/epel-6-i386/glibc-2.17-55.fc20/
-if you have x86_64 go here   https://download.copr.fedorainfracloud.org/results/mosquito/myrepo-el6/epel-6-x86_64/glibc-2.17-55.fc20/
-Now just just right click and copy the first link adress then enter putty with root and follow example , in my case  i686:
wget https://download.copr.fedorainfracloud.org/results/mosquito/myrepo-el6/epel-6-i386/glibc-2.17-55.fc20/glibc-2.17-55.el6.i686.rpm 
-then install it with : 
sudo rpm -Uvh glibc-2.17-55.el6.i686.rpm \--force --nodeps
-Do this 2 steps for every link separately then u can verify again your glibc version 
ldd --version    and it should look like this:
ldd (GNU libc) 2.17    
then you can restart your server and the glibc error should be gone.

Thanks h3o66 I couldn't do this without him 
Also can you please write the name of the topic as "glib requirements not met error" so that it would be easier to find for those who have this problem.

**Comment by h3o66** [comment]
SOLVED so closing this issue
